json_encoder.go

```diff
@@ -168,6 +172,46 @@ func (enc *jsonEncoder) AddMarshaler(key string, obj LogMarshaler) error {
 	return err
 }
 
+func (enc *jsonEncoder) addArrayBegin() {
+	enc.bytes = append(enc.bytes, '[')
+}
+
+func (enc *jsonEncoder) addArrayEnd() {
+	enc.bytes = append(enc.bytes, ']')
+}
+
+func (enc *jsonEncoder) addArraySep(i int) {
+	if i == 0 {
+		return
+	}
+	enc.bytes = append(enc.bytes, ',')
+}
+
+// AddInts adds a string key and integer slice to the encoder's fields. The key
+// is JSON-escaped.
+func (enc *jsonEncoder) AddInts(key string, vals []int) {
+	enc.addKey(key)
+	enc.addArrayBegin()
+
+	for i, val := range vals {
+		enc.addArraySep(i)
+		enc.bytes = strconv.AppendInt(enc.bytes, int64(val), 10)
```

Comment:
ideally `AddInt` and `AddInts` would share a kernel just like `AddString` and `AddStrings` do.